tests/Core/Content/Product/SalesChannel/Listing/ProductListingRouteTest.php

```diff
@@ -88,4 +96,68 @@ public function testWillUseOriginalInCaseRequestComesFromHomepage(Request $reque
         $this->getOriginal()->expects($this->once())->method('load');
         $this->call($productRoute, $request, $salesChannelContextMock, '2');
     }
+
+    public function testCategoryWithManualProductSelectionAddsCategoryFilter(): void
+    {
+        $categoryId = '69';
+
+        $salesChannelContextMock = $this->getMockedSalesChannelContext(true, '1');
+        $request = Request::create('http://your-shop.de/some-category');
+        $request->setSession($this->getSessionMock());
+
+        $expectedFilter = new EqualsFilter('product.categoriesRo.id', $categoryId);
+        $criteria = new Criteria();
+
+        $this->setCategoryMock($categoryId);
+
+        $productRoute = $this->getRoute();
+        /** @var ProductListingRouteResponse $response */
+        $response = $this->call($productRoute, $request, $salesChannelContextMock, $categoryId, $criteria);
+
+        $this->assertTrue($response->getResult()->getCriteria()->hasEqualsFilter('product.categoriesRo.id'));
+    }
+
+    public function testCategoryWithDynamicProductGroupAddsStreamIdAndFilters(): void
+    {
+        $supportsProductStreams = defined(
+            '\Shopware\Core\Content\Category\CategoryDefinition::PRODUCT_ASSIGNMENT_TYPE_PRODUCT_STREAM'
+        );
+        if (!$supportsProductStreams) {
+            $this->markTestSkipped('Dynamic product groups for categories where introduced in Shopware 6.3.1.0');
+        }
+
+        $categoryId = '69';
+        $streamId = '96';
+
+        $salesChannelContextMock = $this->getMockedSalesChannelContext(true, '1');
+        $request = Request::create('http://your-shop.de/some-category');
+        $request->setSession($this->getSessionMock());
+
+        $criteria = new Criteria();
+        $expectedStockFilter = new EqualsFilter('product.stock', 30);
+        $expectedManufacturerFilter = new EqualsFilter('product.manufacturer.id', 10);
+        $expectedFilters = [$expectedStockFilter, $expectedManufacturerFilter];
+
+        $this->setCategoryMock(
+            $categoryId,
+            CategoryDefinition::PRODUCT_ASSIGNMENT_TYPE_PRODUCT_STREAM,
+            $streamId
+        );
+
+        $this->productStreamBuilderMock->expects($this->once())
+            ->method('buildFilters')
+            ->willReturn($expectedFilters);
+
+        $productRoute = $this->getRoute();
+        /** @var ProductListingRouteResponse $response */
+        $response = $this->call($productRoute, $request, $salesChannelContextMock, $categoryId, $criteria);
+
+        if (method_exists($response->getResult(), 'getStreamId')) {
```

Comment:
Shouldn't this exists, when running with `supportsProductStreams`?

Reply:
There is a comment regarding that in code:
![image](https://user-images.githubusercontent.com/51320851/158174038-9994f36c-f668-4b38-92be-cb703e32fc03.png)

Product Streams were supported before those getters/setters existed